csrc/quantization/cutlass_w8a8/moe/moe_data.cu

```diff
@@ -109,19 +109,17 @@ void get_cutlass_moe_mm_data_caller(
     torch::Tensor& problem_sizes1, torch::Tensor& problem_sizes2,
     torch::Tensor& input_permutation, torch::Tensor& output_permutation,
     const int64_t num_experts, const int64_t n, const int64_t k,
-    const std::optional<torch::Tensor>& blockscale_offsets) {
+    const std::optional<torch::Tensor>& blockscale_offsets,
+    bool force_no_swap) {
   auto stream = at::cuda::getCurrentCUDAStream(topk_ids.device().index());
   auto options_int32 =
       torch::TensorOptions().dtype(torch::kInt32).device(topk_ids.device());
   torch::Tensor atomic_buffer = torch::zeros(num_experts, options_int32);
 
   int num_threads = min(THREADS_PER_EXPERT, topk_ids.numel());
 
-  // Swap-AB should be disabled for FP4 path
-  bool may_swap_ab = (!blockscale_offsets.has_value()) &&
-                     (topk_ids.numel() <= SWAP_AB_THRESHOLD);
-
-  if (may_swap_ab) {
+  bool swap_ab = !force_no_swap && topk_ids.numel() <= SWAP_AB_THRESHOLD;
+  if (swap_ab) {
```

Comment:
If we are not using `blockscale_offsets.has_value()`, we should make sure that nvfp4 path will pass in the `force_no_swap`